### ISSUE TITLE
Weird Behavior of Double's round Method

### ISSUE THREAD

**Issue by cheeseng** [scala-native/scala-native#1071]
This behavior is observed when we're working on scala-native in scalatest, the following code:

```
val a: Double = 4228438087383875356545203991520.000000
a.round
```

In JVM, the code above returns:

```
9223372036854775807
```

but in scala-native, it returns:

```
-9223372036854775808.000000
```

**Comment by LeeTibbert** [comment]
PR 1298 fixes part of this problem. After that fix, the S-N result is still negative but it
no longer has a decimal point and decimal digits.  Progress...

**Comment by LeeTibbert** [comment]
@cheeseng 

Thank you for taking the time to file this report.  Investigating & finally fixing it lead me to
a number of other issues with Math.scala. Better code for all.

Both parts of the issue you reported should be fixed once PR #1306 gets accepted.
That PR should also solve problems with large negative values, NaN, and not properly
rounding ties towards positive Infinity.  Float round() had similar problems, with different
numerical values.

Lee

**Comment by densh** [comment]
Fixed in #1306.
